### PR TITLE
Add Support for autofill with textContentType

### PR DESCRIPTION
**Pull request by Evsio0n**
textContentType **implentation** need to be updated in text_input.dart in Flutter SDK

### CODE REVIEW

**Comment by Evsio0n** [comment]
@googlebot I signed it!

**Comment by Evsio0n** [comment]
Some error, wait a minute.

**Comment by ArtemMartus** [comment]
So how's autofill implementation going?

File: shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm [hunk 1]
```diff
@@ -149,14 +156,22 @@ @interface FlutterTextInputView : UIView <UITextInput>
...
+
 
 // UITextInputTraits
 @property(nonatomic) UITextAutocapitalizationType autocapitalizationType;
 @property(nonatomic) UITextAutocorrectionType autocorrectionType;
 @property(nonatomic) UITextSpellCheckingType spellCheckingType;
 @property(nonatomic) BOOL enablesReturnKeyAutomatically;
 @property(nonatomic) UIKeyboardAppearance keyboardAppearance;
-@property(nonatomic) UIKeyboardType keyboardType;
+@property(nonatomic) UIKeyboardType keyboardType
```
Comment (ArtemMartus) [inline shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm]: I think that line needs semicolon

Comment (Evsio0n) [reply shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm]: Already fixed in blow

File: shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm [hunk 2]
```diff
@@ -45,6 +45,13 @@ static UITextAutocapitalizationType ToUITextAutoCapitalizationType(NSDictionary*
   return UITextAutocapitalizationTypeNone;
 }
 
+@available iOS (10.0,*)
+static UITextContentType ToUITextContentType(NSDictionary* type){
+      NSString* contentType = type[@"textType"];
+    if ([contentType isEqualToString:@"TextContent.username"])
+        return UITextContentTypeUsername;
```
Comment (justinmc) [inline shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm]: The above 3 lines have indentation problems.  Indentation should be 2 spaces.

File: shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm [hunk 3]
```diff
@@ -45,6 +45,13 @@ static UITextAutocapitalizationType ToUITextAutoCapitalizationType(NSDictionary*
   return UITextAutocapitalizationTypeNone;
 }
 
+@available iOS (10.0,*)
+static UITextContentType ToUITextContentType(NSDictionary* type){
```
Comment (justinmc) [inline shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm]: Missing a space, `){` should be `) {`.

File: shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm [hunk 4]
```diff
@@ -149,6 +156,14 @@ @interface FlutterTextInputView : UIView <UITextInput>
 @property(nonatomic, strong) UITextRange* markedTextRange;
 @property(nonatomic, copy) NSDictionary* markedTextStyle;
 @property(nonatomic, assign) id<UITextInputDelegate> inputDelegate;
+/**
+ This file is to make ios 10 or higher device using autofill
+ 
+ Reference to  github.com/flutter/flutter/issues/33040
+ **/
+if(@available(10.0,0))
+@property(nonatomic) UITextContentType textContentType;
```
Comment (justinmc) [inline shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm]: Is this compiling and working properly?  I was recently having trouble with getting the syntax right for specifying the version of iOS like this in https://github.com/flutter/engine/pull/13863.  I thought it should be:

```dart
@property(nonatomic) UITextContentType textContentType API_AVAILABLE(ios(10.0));
```

Comment (Evsio0n) [reply shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm]: I will check it now. Sorry for none testing.

**Comment by ArtemMartus** [comment]
@googlebot I consent

**Comment by Evsio0n** [comment]
#13878 https://github.com/flutter/flutter/issues/33040

**Comment by ArtemMartus** [comment]
@googlebot I fixed it.

**Comment by LongCatIsLooong** [comment]
@Evsio0n it looks like these two commits: https://github.com/flutter/engine/pull/13878/commits/785699a6fc62a3a4a8f5323813ea3f09fa8cf9b0 and https://github.com/flutter/engine/pull/13878/commits/7ca9e4ee498f0677fdca74e55b24748565fe3333 are created under a different email address (that hasn't signed the CLA). Could you [rebase these commits](https://github.com/flutter/engine/pull/13878/commits/7ca9e4ee498f0677fdca74e55b24748565fe3333) using the email address that you signed the CLA with?

**Comment by Evsio0n** [comment]
> @Evsio0n it looks like these two commits: [785699a](https://github.com/flutter/engine/commit/785699a6fc62a3a4a8f5323813ea3f09fa8cf9b0) and [7ca9e4e](https://github.com/flutter/engine/commit/7ca9e4ee498f0677fdca74e55b24748565fe3333) are created under a different email address (that hasn't signed the CLA). Could you [rebase these commits](https://stackoverflow.com/questions/3042437/how-to-change-the-commit-author-for-one-specific-commit) using the email address that you signed the CLA with?

**Comment by Evsio0n** [comment]
> @Evsio0n it looks like these two commits: [785699a](https://github.com/flutter/engine/commit/785699a6fc62a3a4a8f5323813ea3f09fa8cf9b0) and [7ca9e4e](https://github.com/flutter/engine/commit/7ca9e4ee498f0677fdca74e55b24748565fe3333) are created under a different email address (that hasn't signed the CLA). Could you [rebase these commits](https://stackoverflow.com/questions/3042437/how-to-change-the-commit-author-for-one-specific-commit) using the email address that you signed the CLA with?

Fixed, Thanks.

**Comment by cbenhagen** [comment]
@Evsio0n you might also want to fix the formatting issues mentioned in https://cirrus-ci.com/task/6514581622489088

File: shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm [hunk 5]
```diff
@@ -191,6 +205,8 @@ - (instancetype)init {
     _enablesReturnKeyAutomatically = NO;
     _keyboardAppearance = UIKeyboardAppearanceDefault;
     _keyboardType = UIKeyboardTypeDefault;
+    if(@available iOS (11.0,*))
```
Comment (cbracken) [inline shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm]: The syntax above should be: `if (@available(iOS 11.0, *))`

File: shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm [hunk 6]
```diff
@@ -754,6 +770,8 @@ - (void)hideTextInput {
 
 - (void)setTextInputClient:(int)client withConfiguration:(NSDictionary*)configuration {
   NSDictionary* inputType = configuration[@"inputType"];
+  if(@available iOS (11.0,*))
```
Comment (cbracken) [inline shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm]: The syntax above should be: `if (@available(iOS 11.0, *))`

File: shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm [hunk 7]
```diff
@@ -762,6 +780,8 @@ - (void)setTextInputClient:(int)client withConfiguration:(NSDictionary*)configur
   }
 
   _activeView.keyboardType = ToUIKeyboardType(inputType);
+  if(@available iOS (11.0,*))
```
Comment (cbracken) [inline shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm]: The syntax above should be: `if (@available(iOS 11.0, *))`

File: shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm [hunk 8]
```diff
@@ -754,6 +770,8 @@ - (void)hideTextInput {
 
 - (void)setTextInputClient:(int)client withConfiguration:(NSDictionary*)configuration {
   NSDictionary* inputType = configuration[@"inputType"];
+  if(@available iOS (11.0,*))
+      NSDictionary* contentType = configuration[@"contentType"];
```
Comment (cbracken) [inline shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm]: It looks like this is an unused variable.

Comment (Evsio0n) [reply shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm]: This variable is referred by https://github.com/flutter/flutter/issues/33040

Comment (Evsio0n) [reply shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm]: > It looks like this is an unused variable.

Maybe I will check CI After Exam Month

**Comment by cbracken** [comment]
Thanks for your contribution. Since we haven't seen any further progress on this, and it currently does not compile, we're going to reluctantly close this PR for now, but if you get time to address the comments, we'd be more than happy to take a look at a new PR. If you do send a new PR, it would be very useful to reference this one in the description in order to help reviewers see the previous context.

**Pull request closed without merging**